Sources/Amplitude/Events/EventOptions.swift

```diff
@@ -12,7 +12,7 @@ public class EventOptions {
     public var deviceId: String?
     public var timestamp: Int64?
     public var eventId: Int64?
-    public var sessionId: Int64? = -1
+    public var sessionId: Int64 = -1
```

Comment:
Thanks @falconandy, I think this should be updated to allow `options.sessionId` to take precedence over `event.sessionId` to keep in line with other EventOptions overriding Event values. In this case I think we will need to keep the nullability. wdyt?

```
sessionId = eventOptions.sessionId != nil ? eventOptions.sessionId : sessionId
```

cc @yuhao900914

Reply:
Event wins in both Kotlin/Swift SDK for all EventOptions properties including `session_id` - so ` to keep in line with other EventOptions overriding Event values.` doesn't work for the SDKs.

Event wins:
https://github.com/amplitude/Amplitude-Swift/blob/main/Sources/Amplitude/Events/BaseEvent.swift#L150
https://github.com/amplitude/Amplitude-Kotlin/blob/main/core/src/main/java/com/amplitude/core/events/BaseEvent.kt#L13

EventOptions wins:
https://github.com/amplitude/Amplitude-Python/blob/main/src/amplitude/event.py#L545
https://github.com/amplitude/analytics-go/blob/main/amplitude/client.go#L69
https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-core/src/utils/event-builder.ts#L20

If we want to change precedence in Kotlin/Swift SDKs for all EventOptions properties - it could be a kind of breaking change. WDYT?

Reply:
Ampli relies on EventOptions to override default values if needed by the user. 

That is a bug in Swift & Kotlin. They should be updated to match the other SDKs.

As far as breaking change:
1. Can you please check the logic and make sure it won't affect anything internal to the SDK.
2. SDK is in Beta and I don't think too many people are using EventOptions right now, so not very concerned on it affecting customers.
3. I would rather call this a "Bugfix" than a breaking change since it wasn't aligned with the functionality in other SDKs.

Reply:
Made a new ticket for Kotlin https://amplitude.atlassian.net/browse/AMP-75272

Reply:
Fixed, added a couple of tests